libandroid-navigation-ui/src/test/java/com/mapbox/services/android/navigation/ui/v5/NavigationViewSubscriberTest.java

```diff
@@ -0,0 +1,79 @@
+package com.mapbox.services.android.navigation.ui.v5;
+
+import android.arch.lifecycle.Lifecycle;
+import android.arch.lifecycle.LifecycleOwner;
+import android.arch.lifecycle.MutableLiveData;
+import android.arch.lifecycle.Observer;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class NavigationViewSubscriberTest {
+
+  @Mock
+  private LifecycleOwner lifecycleOwner;
+  @Mock
+  private Lifecycle lifecycle;
+  @Mock
+  private NavigationPresenter navigationPresenter;
+  @Mock
+  private NavigationViewModel navigationViewModel;
+
+  @Rule
+  public MockitoRule mockitoRule = MockitoJUnit.rule();
+
+  private NavigationViewSubscriber theNavigationViewSubscriber;
+
+  @Before
+  public void setup() {
+    when(lifecycleOwner.getLifecycle()).thenReturn(lifecycle);
+
+    theNavigationViewSubscriber = new NavigationViewSubscriber(lifecycleOwner,
+      navigationViewModel, navigationPresenter);
+  }
+
+  @Test
+  public void checkLifecycleObserverAddedWhenCreateSubscriber() {
+    verify(lifecycleOwner.getLifecycle()).addObserver(theNavigationViewSubscriber);
+  }
+
+  @Test
+  public void checkObserversAreRemovedWhenUnsubscribe() {
+    when(navigationViewModel.retrieveRoute()).thenReturn(mock(MutableLiveData.class));
+    when(navigationViewModel.retrieveNavigationLocation()).thenReturn(mock(MutableLiveData.class));
+    when(navigationViewModel.retrieveDestination()).thenReturn(mock(MutableLiveData.class));
+    when(navigationViewModel.retrieveShouldRecordScreenshot()).thenReturn(mock(MutableLiveData.class));
+
+    theNavigationViewSubscriber.unsubscribe();
+
+    verify(navigationViewModel.retrieveRoute()).removeObservers(eq(lifecycleOwner));
```

Comment:
Generally is preferable having only 1 assertion per test 👀 https://speakerdeck.com/guardiola31337/elegant-unit-testing-mobilization-2016?slide=57

The problem of having multiple assertions are:
- It's annoying to have to look at the stacktrace to know what's failing (which assertion failed?)
- You don't have information about the remaining assertions (because when an assertion fails it breaks the execution of the test)

That's why is preferable to have only 1 assertion per test and one test per scenario. Basically, if one of them fails you'll know at a glance what in the code might have caused that failure and in which part (class, method, etc.).